core/src/main/java/com/sequenceiq/cloudbreak/service/stack/flow/TerminationService.java

```diff
@@ -114,12 +114,18 @@ private void deleteOnlyIfForced(Stack stack, boolean force, String terminatedNam
     }
 
     private void updateToDeleteCompleted(Stack stack, String terminatedName, String statusReason) {
-        stack.setName(terminatedName);
-        stack.setTerminated(clock.getCurrentTimeMillis());
-        stackService.save(stack);
-        stackUpdater.updateStackStatus(stack.getId(), DetailedStackStatus.DELETE_COMPLETED, statusReason);
-        if (stack.getType().equals(StackType.WORKLOAD)) {
-            grpcUmsClient.notifyResourceDeleted(stack.getResourceCrn(), MDCUtils.getRequestId());
+        try {
+            transactionService.required(() -> {
+                stack.setName(terminatedName);
+                stack.setTerminated(clock.getCurrentTimeMillis());
+                stackService.save(stack);
+                stackUpdater.updateStackStatus(stack.getId(), DetailedStackStatus.DELETE_COMPLETED, statusReason);
+                if (stack.getType().equals(StackType.WORKLOAD)) {
+                    grpcUmsClient.notifyResourceDeleted(stack.getResourceCrn(), MDCUtils.getRequestId());
```

Comment:
are you sure you want to include an external call in the transaction? I'm against it

Reply:
oh god! :) thanks..